lib/types.nix

```diff
@@ -395,10 +395,12 @@ rec {
       merge = mergeEqualOption;
     };
 
-    listOf = elemType: mkOptionType rec {
+    # NixOS apparently has a different notion of "type" than nix(lang)
+    # does; see discussion in GH #173568
```

Comment:
Is this the type of comment you would like to have in your code base?
It is accusatory and it lacks real meaning. Can't even paste a URL. A few extra words could have fixed that. (and a lazy Ctrl+L Ctrl+C to copy the whole url)

(I'll shut up now, but I figured this is useful feedback for your future PRs)

Reply:
I would be happy to include complete URLs, and will do so in the future.  It wasn't laziness.  Some people get cranky about big long URLs messing up the line-wrapping in multi-line comments.  Other people insist on prefixing issue numbers to identify the bugtracker (BZ for Bugzilla, etc) in case the project migrates from one to another in the future.  Personally, I don't have strong feelings about any of these things, and just go along with the prevailing trend.  Now that I know, my future PRs which reference github issues from a comment will use a complete URL.

Regarding the part of the comment before the semicolon:

> Is this the type of comment you would like to have in your code base?

It is highly non-obvious that `types.nix` uses a different definition of "list" than the Nix language definition does.  I do think that non-obvious departures like that need to be documented with comments.